### PR TITLE
BAH-2281 | Starting a visit/encounter for a patient

### PR DESCRIPTION
**Pull request by abinaya-u**
## Requirements

- [x] This PR has a proper title that briefly describes the work done
- [x] I have squashed / amended the comments to make it more readable
- [x] I have included link to all the JIRA ticket('s)
- [x] I wrote the code as a pair or at least performed a self-review of my own code
- [x] I have updated the documentation on [Bahmni Wiki](https://bahmni.atlassian.net/wiki/spaces/BAH/overview) or [README](https://github.com/Bahmni/bahmni-lab-frontend/blob/main/README.md) (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary
Wanted to link the lab report with an encounter, So that the reports are systematically associated with encounters 

## JIRA tickets
https://bahmni.atlassian.net/browse/BAH-2281?atlOrigin=eyJpIjoiMTIwNzhiZDVhMjM3NGI0Mzk3YTY4NTg3NjMxMjM2ZDgiLCJwIjoiaiJ9

### CODE REVIEW

File: src/home/home.tsx [hunk 1]
```diff
@@ -26,6 +28,38 @@ const Home = () => {
...
+  const {data: encounterTypeResponse} = useSWR(encounterTypeUrl, fetcher, {
+    revalidateIfStale: false,
+    revalidateOnFocus: false,
+    revalidateOnReconnect: false,
+  })
+
+  useEffect(() => {
+    if (configResponse && !localStorage.getItem('defaultVisitType')) {
+      localStorage.setItem(
+        'defaultVisitType',
```
Comment (sivareddyp) [inline src/home/home.tsx]: can you extract `defaultVisitType` to a const variable ?

Comment (abinaya-u) [reply src/home/home.tsx]: sure @sivareddytw

File: src/patient-lab-dashboard/upload-report/upload-report.resources.tsx [hunk 2]
```diff
@@ -138,3 +147,69 @@ const removeBase64 = fileData => {
...
+      locationUuid,
+      patientUuid,
+      encounterUuid: null,
+      visitUuid: null,
+      providers: [
+        {
+          uuid: providerUuid,
+        },
+      ],
+      visitType: 'OPD',
```
Comment (sivareddyp) [inline src/patient-lab-dashboard/upload-report/upload-report.resources.tsx]: can this be configured ?

File: src/patient-lab-dashboard/upload-report/upload-report.resources.tsx [hunk 3]
```diff
@@ -138,3 +147,69 @@ const removeBase64 = fileData => {
...
+      encounterUuid: null,
+      visitUuid: null,
+      visitType: 'OPD',
+      encounterTypeUuid: getEncounterTypeUuid(
+        selectedTest,
+        selectedPendingOrder,
+      ),
+    }
+  }
+  return postApiCall(bahmniEncounterUrl, requestBody, ac)
```
Comment (sivareddyp) [inline src/patient-lab-dashboard/upload-report/upload-report.resources.tsx:196]: is this handling API failure scenario also ?

File: src/utils/api-utils.ts [hunk 4]
```diff
@@ -18,14 +18,20 @@ export const getLabTests = () =>
...
 
 export const saveDiagnosticReportURL = '/ws/fhir2/R4/DiagnosticReport'
 
 export const auditLogURL = '/ws/rest/v1/auditlog'
 
 export const auditLogGlobalPropertyURL =
-  '/ws/rest/v1/bahmnicore/sql/globalproperty?property=bahmni.enableAuditLog'
+'/ws/rest/v1/bahmnicore/sql/globalproperty?property=bahmni.enableAuditLog'
+
+export const configUrl = '/../bahmni_config/openmrs/apps/registration/app.json'
```
Comment (sivareddyp) [inline src/utils/api-utils.ts:30]: do we need starting `/` ?

File: src/patient-lab-dashboard/upload-report/upload-report.resources.tsx (hunk 2 above)
Comment (gsluthra) [reply src/patient-lab-dashboard/upload-report/upload-report.resources.tsx]: We can't hardcode visit type. All visit types are a configuration which implementers set. Have to have a configuration for this value.

Comment (abinaya-u) [reply src/patient-lab-dashboard/upload-report/upload-report.resources.tsx]: configuration should come from app config which is fetched initially and stored in localStorage. I will make the changes to get the visitType value from localStorage

File: src/utils/api-utils.ts (hunk 4 above)
Comment (abinaya-u) [reply src/utils/api-utils.ts:30]: it works only if the string has `/` at the beginning

File: src/home/home.tsx (hunk 1 above)
Comment (abinaya-u) [reply src/home/home.tsx]: changes have been made

File: src/patient-lab-dashboard/upload-report/upload-report.resources.tsx (hunk 3 above)
Comment (abinaya-u) [reply src/patient-lab-dashboard/upload-report/upload-report.resources.tsx:196]: incorporated the feedback

File: src/home/home.tsx [hunk 5]
```diff
@@ -26,6 +33,41 @@ const Home = () => {
     revalidateOnReconnect: false,
   })
 
+  const {data: configResponse} = useSWR(configUrl, fetcher, {
+    revalidateIfStale: false,
```
Comment (sivareddyp) [inline src/home/home.tsx]: can we have a variable for SWR options & use it all the places?

Comment (abinaya-u) [reply src/home/home.tsx]: changes have been made, extracted to a common place

**Review by sivareddyp — APPROVED** [review]

**Pull request merged**